Microsoft.Toolkit.Uwp.UI.Controls/Loading/Loading.cs

```diff
@@ -0,0 +1,158 @@
+﻿using System;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+using Windows.UI.Xaml.Media.Animation;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// Loading control allows to show an loading animation with some xaml in it.
+    /// </summary>
+    public sealed partial class Loading : ContentControl
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="Loading"/> class.
+        /// </summary>
+        public Loading()
+        {
+            this.HorizontalContentAlignment = HorizontalAlignment.Stretch;
+            this.VerticalContentAlignment = VerticalAlignment.Stretch;
+
+            LoadingVerticalAlignment = VerticalAlignment.Stretch;
+            LoadingHorizontalAlignment = HorizontalAlignment.Stretch;
+
+            RootGrid = new Grid();
+            BackgroundGrid = new Grid();
+            ContentGrid = new Grid();
+            RootGrid.Children.Add(BackgroundGrid);
+            RootGrid.Children.Add(ContentGrid);
+            Content = RootGrid;
+        }
+
+        private void CreateLoadingControl()
+        {
+            if (IsLoading)
+            {
+                ContentGrid.Children.Clear();
+                if (LoadingBackground == null && LoadingOpacity == 0d)
+                {
+                    BackgroundGrid = null;
+                }
+                else
+                {
+                    BackgroundGrid.Background = LoadingBackground;
+                    BackgroundGrid.Opacity = LoadingOpacity;
+                }
+
+                CreateStoryboard(translateBegin: 40d, translateEnd: 0d, opacityBegin: 0d, opacityEnd: 1d);
+                Animation.Begin();
+
+                var contentControl = LoadingContent?.LoadContent() as FrameworkElement;
+                if (contentControl == null)
+                {
+                    return;
+                }
+
+                contentControl.HorizontalAlignment = LoadingHorizontalAlignment;
+                contentControl.VerticalAlignment = LoadingVerticalAlignment;
+
+                ContentGrid.Children.Add(contentControl);
+            }
+            else
+            {
+                CreateStoryboard(translateBegin: 0d, translateEnd: 40d, opacityBegin: 1d, opacityEnd: 0d);
+                Animation.Begin();
+            }
+        }
+
+        private void CreateStoryboard(double translateBegin, double translateEnd, double opacityBegin, double opacityEnd)
+        {
+            Animation = new Storyboard();
```

Comment:
These animations should be created in xaml in the control's default style and triggered using VisualStates. This allows devs to retemplate the control if they need to and change some of these values.